core/app/models/spree/promotion/rules/first_purchase_since.rb

```diff
@@ -0,0 +1,36 @@
+module Spree
+  class Promotion
+    module Rules
+      class FirstPurchaseSince < PromotionRule
```

Comment:
Feel free to ignore, but something like `FirstRepeatPurchaseSince` would make the intent of this more immediately clear to me (since it only applies if you've previously made a purchase).